runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/graph/CreateExecutableStageNodeFunction.java

```diff
@@ -191,6 +195,37 @@ public Node apply(MutableNetwork<Node, Edge> input) {
         Iterables.filter(input.nodes(), InstructionOutputNode.class)) {
       InstructionOutput instructionOutput = node.getInstructionOutput();
 
+      // If this is the input PCollection or the output PCollection for an ExecutableStage, it's
+      // necessary to check whether the window coder is not a GlobalWindow coder.
+      if (isExecutableStageInputPCollection(input, node)
+          || isExecutableStageOutputPCollection(input, node)) {
```

Comment:
for education purpose, what are the pcollection types beyond Input and output?

Reply:
As for an ExecutableStage, there are 3 types of PCollection from my knowledge. The ```input PCollection``` is the PCollection flowing from runner to sdk. The ```output PCollection``` is the PCollection flowing from sdk to runner. The other PCollections are the intermediate PCollection between 2 PTransforms.